lib/ciphersuite.ml

```diff
@@ -37,22 +38,31 @@ type aead_cipher =
   | AES_256_GCM
   [@@deriving sexp]
 
-type payload_protection =
-  | Stream of stream_cipher * Nocrypto.Hash.hash
-  | Block of block_cipher * Nocrypto.Hash.hash
-  | AEAD of aead_cipher
-  [@@deriving sexp]
+(* this is K_LEN, max 8 N_MIN from RFC5116 -- as defined in TLS1.3 spec *)
```

Comment:
(and would it make sense to mark this as `1.3`-only like `payload_protection13` below?)
(( wondering if it might be clearer to put the 1.3-functionality in a module here so it becomes more clear what is used where))